dmriprep/runtime.py

```diff
@@ -0,0 +1,239 @@
+               and os.path.isdir(os.path.join(path, sub_dir, ses_dir)):
+                sub_ses_dirs.append(os.path.join(path, sub_dir, ses_dir))
+    return sub_ses_dirs
+
+
+def run(datadir, outdir, simg_file, name="dmriprep",
+        process=False, njobs=10, use_pbs=False, test=False):
+    """ Parse data and execute the processing with hopla.
+
+    Parameters
+    ----------
+    datadir: str
+        path to the BIDS rawdata directory.
+    outdir: str
+        path to the BIDS derivatives directory.
+    name: str, default 'dmriprep'
+        the name of the cirrent analysis.
+    process: bool, default False
+        optionnaly launch the process.
+    njobs: int, default 10
+        the number of parallel jobs.
+    use_pbs: bool, default False
+        optionnaly use PBSPRO batch submission system.
+    cmd: str, default 'limri'
+        the command to execute.
+    test: bool, default False
+        optionnaly, select only one subject.
+    """
+    list_sub_ses = get_sub_ses(datadir)
+    list_dwi, list_bvec, list_bval, list_pe, list_readout, list_outdir = \
+        [], [], [], [], [], []
+    for sub_ses in list_sub_ses:
+        # Verify all the mandatory file for prequal launch
+        # dwi
+        _dwi = glob.glob(os.path.join(sub_ses, "dwi",
+                                      "*_acq-DWI*_run-*_dwi.nii.gz"))
+        _dwi.sort()
+        if len(_dwi) != 2:
+            print(f"this sub and session don't have 2 dwi.nii.gz : "
+                  f"'{sub_ses}'")
+            continue
+        dwi_files = ",".join(_dwi)
+
+        # bvec
+        _bvec = glob.glob(os.path.join(sub_ses, "dwi",
+                                       "*_acq-DWI*_run-*_dwi.bvec"))
+        _bvec.sort()
+        if len(_bvec) != 2:
+            print(f"this sub and session don't have 2 .bvec : "
+                  f"'{sub_ses}'")
+            continue
+        bvec_files = ",".join(_bvec)
+
+        # bval
+        _bval = glob.glob(os.path.join(sub_ses, "dwi",
+                                       "*_acq-DWI*_run-*_dwi.bval"))
+        _bval.sort()
+        if len(_bval) != 2:
+            print(f"this sub and session don't have 2 .bval : "
+                  f"'{sub_ses}'")
+            continue
+        bval_files = ",".join(_bval)
+        # json
+        _json = glob.glob(os.path.join(sub_ses, "dwi",
+                                       "*_acq-DWI*_run-*_dwi.json"))
+        _json.sort()
+
+        if len(_json) != 2:
+            print(f"this sub and session don't have 2 .json : "
+                  f"'{sub_ses}'")
+            continue
+
+        # PhaseEncodingAxis and EstimatedTotalReadoutTime
+        _pe = []
+        _readout = []
+        error_flag = False
+        for file in _json:
+            with open(file, 'r') as json_file:
+                try:
+                    # load the JSON data from the file
+                    data = json.load(json_file)
+                except json.decoder.JSONDecodeError as e:
```

Comment:
Just raise an error as data will not be affected and the rest of the code do not handles this situation.